lib/rucio/core/replica.py

```diff
@@ -2392,32 +2534,64 @@ def touch_replica(replica, *, session: "Session"):
     try:
         accessed_at, none_value = replica.get('accessed_at') or datetime.utcnow(), None
 
-        session.query(models.RSEFileAssociation).\
-            filter_by(rse_id=replica['rse_id'], scope=replica['scope'], name=replica['name']).\
-            with_hint(models.RSEFileAssociation, "index(REPLICAS REPLICAS_PK)", 'oracle').\
-            with_for_update(nowait=True).one()
-
-        stmt = update(models.RSEFileAssociation).\
-            filter_by(rse_id=replica['rse_id'], scope=replica['scope'], name=replica['name']).\
-            prefix_with("/*+ index(REPLICAS REPLICAS_PK) */", dialect='oracle').\
-            execution_options(synchronize_session=False).\
-            values(accessed_at=accessed_at,
-                   tombstone=case((and_(models.RSEFileAssociation.tombstone != none_value,
-                                        models.RSEFileAssociation.tombstone != OBSOLETE),
-                                   accessed_at),
-                                  else_=models.RSEFileAssociation.tombstone))
+        stmt = select(
+            models.RSEFileAssociation
+        ).with_hint(
+            models.RSEFileAssociation, 'INDEX(REPLICAS REPLICAS_PK)', 'oracle'
+        ).where(
+            and_(models.RSEFileAssociation.rse_id == replica['rse_id'],
+                 models.RSEFileAssociation.scope == replica['scope'],
+                 models.RSEFileAssociation.name == replica['name'])
+        ).with_for_update(
+            nowait=True
+        )
+        session.execute(stmt).one()
+
+        stmt = update(
+            models.RSEFileAssociation
+        ).where(
+            and_(models.RSEFileAssociation.rse_id == replica['rse_id'],
+                 models.RSEFileAssociation.scope == replica['scope'],
+                 models.RSEFileAssociation.name == replica['name'])
+        ).prefix_with(
+            "/*+ INDEX(REPLICAS REPLICAS_PK) */", dialect='oracle'
+        ).values({
+            models.RSEFileAssociation.accessed_at: accessed_at,
+            models.RSEFileAssociation.tombstone: case((models.RSEFileAssociation.tombstone.not_in([OBSOLETE,
```

Comment:
Should this `case` be broke into different lines? It's unwieldy as is.

Reply:
Changed to:

```
            models.RSEFileAssociation.accessed_at: accessed_at,
            models.RSEFileAssociation.tombstone: case(
                (models.RSEFileAssociation.tombstone.not_in([OBSOLETE, none_value]),
                 accessed_at),
                else_=models.RSEFileAssociation.tombstone)
        }).execution_options(
```